openassessment/xblock/data_conversion.py

```diff
@@ -113,14 +124,22 @@ def create_prompts_list(prompt_or_serialized_prompts):
     if prompt_or_serialized_prompts is None:
         prompt_or_serialized_prompts = ''
 
-    try:
-        prompts = json.loads(prompt_or_serialized_prompts)
-    except ValueError:
-        prompts = [
-            {
-                'description': prompt_or_serialized_prompts,
-            }
-        ]
+    if type(prompt_or_serialized_prompts) is dict:
+        # Lazy translation objects cannot be parsed to json normally
+        # https://docs.djangoproject.com/en/1.8/topics/serialization/#serialization-formats-json
+        # we need to pass it with a custom encoder.
+        decode_prompts = json.dumps(prompt_or_serialized_prompts, cls=LazyEncoder)
```

Comment:
@diegomillan Let's rename `decode_prompts` to `serialized_prompts` -- that's more precise and consistent with the naming scheme that this method uses for other variables.